sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -371,7 +371,7 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) {
   // Note that HiveUDFs will be overridden by functions registered in this context.
   @transient
   override protected[sql] lazy val functionRegistry: FunctionRegistry =
-    new OverrideFunctionRegistry(new HiveFunctionRegistry(FunctionRegistry.builtin))
+    new HiveFunctionRegistry(FunctionRegistry.builtin)
```

Comment:
Yes, that's the tricky thing, as this code path is hidden(never be called) as we had the wrapper of `OverrideFunctionRegistry`, see code in HiveContext:
```scala
override protected[sql] lazy val functionRegistry: FunctionRegistry =
    new OverrideFunctionRegistry(new HiveFunctionRegistry(FunctionRegistry.builtin))
```

Since the `OverrideFunctionRegistry` is removed, we should update this part too.